src/main/java/org/icgc/argo/workflow_management/streams/WesConsumerConfig.java

```diff
@@ -71,70 +71,55 @@ public void init() {
   private Disposable createWfMgmtRunMsgForExecuteConsumer() {
     return createTransConsumerStream(rabbit, topicExchangeName, queueName, topicRoutingKeys)
         .receive()
-        .doOnNext(consumeAndExecuteInitializeOrCancel())
-        .onErrorContinue(handleRabbitStreamError())
+        // consume each tx msg and flatMap into publisher of Mono<Void>.
+        // Mono<Void> is used so reactor can manage subscriptions and publisher signals.
+        .flatMap(this::consumeMessageAndExecuteInitializeOrCancel)
+        // Continue on error so disposable doesn't die.
+        // The tx commit/reject is handled in the flatmap so no need to worry about that here
+        .onErrorContinue(catchStreamError())
+        .log(WES_CONSUMER)
         .subscribe();
   }
 
-  private Consumer<Transaction<WfMgmtRunMsg>> consumeAndExecuteInitializeOrCancel() {
-    return tx -> {
-      val msg = tx.get();
-      log.debug("WfMgmtRunMsg received: {}", msg);
-
-      if (msg.getState().equals(RunState.INITIALIZING)) {
-        val runParams = createRunParams(msg);
-        wes.run(runParams)
-            .onErrorContinue(handleWesError(tx))
-            .subscribe(
-                runsResponse -> {
-                  log.info("Initialized: {}", msg);
-                  tx.commit();
-                });
-      } else if (msg.getState().equals(RunState.CANCELING)) {
-        wes.cancel(msg.getRunId())
-            .retryWhen(RetrySpec.backoff(3, Duration.ofMinutes(3)))
-            .onErrorContinue(handleWesError(tx))
-            .subscribe(
-                runsResponse -> {
-                  log.info("Cancelled: {}", runsResponse);
-                  tx.commit();
-                });
-      } else {
-        log.debug("Ignoring: {}", msg);
-        tx.commit();
-      }
-    };
+  private Mono<Void> consumeMessageAndExecuteInitializeOrCancel(Transaction<WfMgmtRunMsg> tx) {
```

Comment:
I'm not too keen on `Mono<Void>` but I understand why, maybe we can make commitTx return something even if we don't use it just right now?